buttercup.el

```diff
@@ -851,7 +863,31 @@ function containing the body instructions passed to `it'."
   (buttercup-suite-add-child buttercup--current-suite
                              (make-buttercup-spec
                               :description description
-                              :function body-function)))
+                              :function
+                               (if (buttercup-suite-environment buttercup--current-suite)
+                                   (let ((suite buttercup--current-suite))
+                                     (lambda ()
+                                       (let ((old
+                                              (mapcar
+                                               (lambda (binding)
+                                                 (if (consp binding)
+                                                     (cons (car binding) (symbol-value (car binding)))
+                                                   (cons binding (symbol-value binding))))
+                                               (buttercup-suite-environment suite))))
+                                         (unwind-protect
+                                             (progn
+                                               (mapc
+                                                (lambda (binding)
+                                                  (if (consp binding)
+                                                      (setf (symbol-value (car binding)) (eval (cadr binding)))
```

Comment:
`eval` :(  But, since these are functions and we are passing data I don't see any good way, other than walking the trees in `describe` macro and looking for `it` forms and installing let-bindings there.